pkgs/servers/web-apps/ackee/default.nix

```
@@ -0,0 +1,73 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, makeWrapper
+, mkYarnPackage
+, nodejs
+}:
+
+mkYarnPackage rec {
+  pname = "ackee";
+  version = "3.1.1";
+
+  src = fetchFromGitHub {
+    owner = "electerious";
+    repo = "Ackee";
+    rev = "v${version}";
+    sha256 = "f02oL36i9CeJz/3PEiAZJQgNhXMuO6/k+k+83jomZdI=";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  packageJSON = ./package.json;
+  yarnLock = "${src}/yarn.lock";
+  yarnNix = ./yarn.nix;
+
+  dontInstall = true;
+
+  prePatch = ''
+    for f in src/types/index.js src/resolvers/index.js; do
+      substituteInPlace $f --replace 'graphql-tools' '@graphql-tools/merge'
+    done
+  '';
+
+  postConfigure = ''
+    rm deps/ackee/node_modules
+    cp -R "$node_modules" deps/ackee
+    chmod -R u+w deps/ackee
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    cd deps/ackee
+    npm run build
+
+    runHook postBuild
+  '';
+
+  distPhase = ''
+    runHook preDist
+
+    mkdir -p $out/bin
+    cp -R {src,dist,node_modules,package.json} $out
+
+    cat > $out/bin/ackee <<EOF
+      #!${stdenv.shell}/bin/sh
+      exec ${nodejs}/bin/node $out/src/index.js
+    EOF
+    chmod +x $out/bin/ackee
+    wrapProgram $out/bin/ackee \
+      --set NODE_PATH "$out/node_modules"
+
```

I think that maybe

```suggestion
    makeWrapper ${nodejs}/bin/node $out/bin/ackee \
      --set NODE_PATH "$out/node_modules" \
      --add-flags "$out/src/index.js"
```

would work for this?

---

Yes, that works, thanks!